test/test_autograd.py

```diff
@@ -4213,7 +4213,9 @@ def add_test(
             test_name = basic_test_name + ''.join('_neg' + str(i) for i, idx in enumerate(dim_perm) if idx < 0)
             if dtype.is_complex:
                 # TODO: remove this. this is temporary while we ramp up the complex support.
-                if name in complex_list and 'scalar' not in test_name and 'constant' not in test_name:
+                if 'complex' in test_name:
```

Comment:
Using just `name` led to accepting all the `tanh` tests and not all of them fail on CUDA i.e. the one with standard `dtype double`.

Have instead added condition to check to filter based on the `test_name`.

Reply:
yup makes perfect sense. I did something similar in this [PR](https://github.com/pytorch/pytorch/pull/38320) as well

Reply:
Should we wait till #38320 is in so that we can use that machinery here. Otherwise I presume, there would be conflict.

Reply:
merged

Reply:
Thanks. 
Done updating.